src/components/member-profile/contribution/index.js

```diff
@@ -63,11 +66,24 @@ const ContributionCard = ({
   urlObj,
 }) => {
   const {
-    task: { title, startedOn, endsOn, status, purpose },
+    task: { id, title, startedOn, endsOn, status, purpose, isNoteworthy },
     prList,
   } = contribution;
   const isTitleAvailable = !!title;
   const featureTitle = isTitleAvailable ? title : prList[0].title;
+  const [showSettings, setShowSettings] = useState(false);
+  const { isOptionKey, setIsOptionKey } = userContext();
+
+  const revealSettingButton = () => {
+    if (isOptionKey) {
+      setShowSettings(true);
+    }
+  };
+
+  const hideSettingButton = () => {
+    setShowSettings(false);
+    setIsOptionKey(false);
+  };
```

Comment:
We will be using the 'option key' quite often. Instead of every component having to write their own handlers with the same kind of code, I am wondering if we could write a reusable logic that can be [_injected_](https://en.wikipedia.org/wiki/Dependency_injection) into the component?

Randhir did something similar in [this PR for reference](https://github.com/Real-Dev-Squad/website-members/pull/410).

Reply:
Okay I will look at it

Reply:
That PR doesn't contain those changes, but will refactor the code in a different PR as dicussed